internal/transport/websocket_server.go

```diff
@@ -115,7 +115,10 @@ func (s *WebsocketServer) start() {
 // ensures that there is at most one reader on a connection by executing all
 // reads from this goroutine.
 func (s *WebsocketServer) readPump() {
-	defer close(s.done)
+	defer func() {
+		close(s.done)
+		s.conn.Close()
+	}()
```

Comment:
Does the order of these matter? I would expect done to be closed last.

Separate but superficially related, maybe consider this format to reduce lines and indentation? :shrug: 
```suggestion
	defer close(s.done)
	defer s.conn.Close()
```

Reply:
Good question!

I've just renamed `done` to `closeWritePump` across the client and server. In the websocket client, only `writePump` is responsible for closing the underlying websockets connection (which is the API gorilla exposes). This additional call to `s.conn.Close()` can be removed. It's likely erroring out anyways. 

The slight issue I have with 
```go
defer close(s.done)
defer s.conn.Close()
```

Is that separate defer statements are added to a LIFO stack - so the reader has to be aware of that when understanding the order the calls are made.

Reply:
> so the reader has to be aware of that when understanding the order the calls are made.

True but that is the typical idiomatic usage and I find it easier to read - I don't know if that is typical though :shrug: 
FWIW separate defers can also make the system more resilient to panics.